File: lib/core/players/interfaces/app_player.dart
import '../player_path.dart';

abstract class AppPlayer {
  Stream<PlayBackEvent> get playbackEventStream;
  Future<bool> loadMedia(String filename);
  Future<void> play();
  Future<void> reset();
  Future<void> pause();
  Future<void> stop();
  Future<void> release();
  Future<bool> isPlaying();
  Future<bool> isReleased();
  Future<Duration> getDuration();
  Future<Duration> getCurrentPosition();
  Future<void> seekTo(Duration position);
  Future<void> setVolume(double volume);
}

File: android/app/src/main/kotlin/com/rytoka/yura/uira/player/interfaces/AppPlayer.kt
package com.rytoka.yura.uira.player.interfaces


interface AppPlayer {
    fun loadMediaFromRaw(fileName: String): Boolean
    fun loadMediaAsset(path: String): Boolean
    fun play()
    fun reset()
    fun pause()
    fun stop()
    fun release()
    fun isPlaying(): Boolean
    fun isReleased(): Boolean
    fun getDuration(): Long
    fun getCurrentPosition(): Long
    fun seekTo(position: Long)
    fun setVolume(volume: Float)
}